### PR TITLE
Final push towards assimilating all add_sibling functionality in generator `siblings`

### PR DESCRIPTION
**Pull request by mih**
- [x] completely dissolve `add-siblings` identical functionality remains available (see pretty much unchanged tests)
- [x] more flexible reconfiguration, with minimized inter-dependencies (before one had to give a URL regardless)
- [x] prepared to inherit more information from parents. there is a single code block in `siblings configure` that has all logic -- it may need to move a bit, but it is all together
- [x] report preferred content configuration
- [x] RF+BF of low-level preferred content API

### CODE REVIEW

File: datalad/distribution/add_sibling.py [hunk 1]
```diff
@@ -99,17 +91,11 @@ class AddSibling(Interface):
                 This option is ignored if there is already a configured sibling
                 dataset under the name given by `name`""",
             constraints=EnsureStr() | EnsureNone()),
-        recursive=recursion_flag,
-        fetch=Parameter(
-            args=("--fetch",),
-            action="store_true",
-            doc="""fetch the sibling after adding"""),
```
Comment (yarikoptic) [inline datalad/distribution/add_sibling.py]: hm, so you are suggesting to remove recursive mode of operation and autofetching upon adding... I think I have used both of those

**Comment by yarikoptic** [comment]
overall -- should I wait a bit longer and consider it for 0.6.0 release or should I proceed with last minute testing and release? @mih?

File: datalad/distribution/add_sibling.py (hunk 1 above)
Comment (mih) [inline datalad/distribution/add_sibling.py]: Nope, I am just slowly disassembling the command for sanity reasons. All functionality remains the same via the `siblings` command.

**Comment by mih** [comment]
This can wait, I am still working on it.

**Comment by mih** [comment]
From my POV this is done. Let's see what the test say.

**Comment by mih** [comment]
Tests look good, let's do one more push.

File: datalad/distribution/add_sibling.py (hunk 1 above)
Comment (mih) [inline datalad/distribution/add_sibling.py]: Now gone completely.

**Comment by mih** [comment]
Continued with preferred content handling -- please stop me!!!

**Comment by bpoldrack** [comment]
LGTM

Waiting for the tests ...

**Pull request merged**